docs/poc2023q1/outline.md

```diff
@@ -0,0 +1,257 @@
+- Rule-based customization of desired state.
+- Propagation of reported state from edge to center.
+- Summarization of reported state in the center.
+- The edge opens connections to the center, not vice-versa.
+
+Some important things that are not attempted in this PoC include the following.
+
+- An implementation that supports a large number of edge clusters.
+- More than one SyncTarget per Location.
+- Return or summarization of status from associated objects (e.g.,
+  ReplicaSet or Pod objects associated with a given Deployment
+  object).
+- A hierarchy with more than two levels.
+- More than baseline security (e.g., HTTPS, Secret objects, bearer
+  token based service authorization).
+
+It is TBD whether the implementation will support intermittent
+connectivity.  This depends on whether we can quickly and easily get a
+syncer that creates the appropriately independent objects in the edge
+cluster and itself tolerates intermittent connectivity.
+
+Also: initially this PoC will not transport non-namespaced objects.
+CustomResourceDefinitions are thus among the objects that will not be
+transported.  Transport of non-namespaced objects may be added after
+the other goals are achieved.
+
+As further matters of roadmapping development of this PoC:
+customization may be omitted at first, and summarization may start
+with only a limited subset of the implicit functionality.
+
+This PoC builds on TMC and makes some compromises to accommodate that.
+The implementation involves workload components (syncers) writing
+status information to inventory objects (SyncTargets).
+
+This PoC compromises on the connection initiative constraint in one
+place: the bootstrap step where central automation connects to edge
+clusters to install EMC machinery.
+
+## Design overview
+
+In very brief: the design is to reduce each edge placement problem to
+many instances of kcp's TMC problem.
+
+See [the overview picture](Edge-PoC-2023q1.svg) for an overview
+picture.
+
+## Inventory Management workspaces
+
+This design takes as a given that something maintains some kcp
+workspaces that contain dynamic collections of Location and SyncTarget
+objects as defined in [kcp
+TMC](https://github.com/kcp-dev/kcp/tree/main/pkg/apis), and that one
+view can be used to read those Location objects and one view can be
+used to read those SyncTarget objects.
+
+To enable the workload management layer to manipulate each edge
+cluster, that cluster's SyncTarget has an annotation that refers to a
+Secret that holds a kubeconfig that can be used by the Bootstrapper to
+install the edge-mc machinery in the edge cluster.  These annotations
+and Secrets are created by the inventory management layer for use by
+the workload management layer.
+
+To complete the plumbing of the syncers, for each SyncTarget the
+mailbox controller creates the following needed accompanying items in
+that SyncTarget's workspace (which must authorize the mailbox
+controller to do this).
+
```

Comment:
Is that the only controller that access the edge cluster?

Reply:
I do not understand why there is a question at this point in the doc regarding edge clusters.

Reply:
Perhaps what I meant by "that SyncTarget's workspace" is not clear.  It is intended to refer to the kcp workspace that contains the SyncTarget object; this is decidedly distinct from the edge cluster described by that SyncTarget.  The original comment here is attached to text talking about that workspace, not the edge cluster.